kubescaler/scaler/aws/cluster.py

```diff
@@ -588,25 +798,39 @@ def delete_cluster(self):
         And let's go backwards - deleting first what we created last.
         """
         logger.info("🔨️ Deleting node workers...")
-        self.delete_workers_stack()
+        if self.eks_nodegroup:
+            self.delete_nodegroup(self.node_group_name)
+        else:
+            self.delete_workers_stack()
         # We could delete keypair, but let's keep for now, assuming could be reused elsewhere
         # and a deletion might be unexpected to the user
 
-        # Now delete the cluster
-        try:
-            self.eks.delete_cluster(name=self.cluster_name)
-        except Exception as e:
-            logger.info(f"⏳️ Cluster likely already deleted: {e}")
-            return
-
+        # An error occurred (ResourceInUseException) when calling the DeleteCluster operation: Cannot delete because cluster <name> currently has an update in progress
+        # TODO Make a good design for this portion
+        while True:
+            try:
+                self.eks.delete_cluster(name=self.cluster_name)
+            except self.eks.exceptions.ResourceInUseException as e:
+                print(f"The cluster resources are busy, retry in a few seconds")
+                time.sleep(60)
+                continue
+            except self.eks.exceptions.ResourceNotFoundException as e:
+                print(f"⏳️ Cluster likely already deleted: {e}")
+                logger.info(f"⏳️ Cluster likely already deleted: {e}")
```

Comment:
You can remove the logger.info line, the print is probably enough. The reason logger.info isn't coming through here is because there is a function in logger.py to set it up (with the level) and I don't think the default is verbose enough. I usually set it up via a client (and we don't have one here).